lib/elixir/lib/record.ex

```diff
@@ -397,35 +389,85 @@ defmodule Record do
       raise ArgumentError, "cannot invoke update style macro inside match"
     end
 
-    keyword = apply_underscore(fields, keyword)
+    case build_updates(keyword, fields, [], [], []) do
```

Comment:
@michalmuskala erl_expand_record has a small heuristic in that, if there are more fields updating than fields being kept, then it does not use setelem at all, and instead allocates a new tuple: https://github.com/erlang/otp/blob/83bf5bea4b872be40253a8f25c9aec3b79d6d66f/lib/stdlib/src/erl_expand_records.erl#L673-L702

Should we have the same?